src/store.c

```diff
@@ -546,7 +566,7 @@ int cs_add(struct clip_store *cs, const char *content, uint64_t *out_hash) {
         *out_hash = hash;
     }
 
-    return cs_snip_add(cs, hash, line, nr_lines);
+    return (ret == 1) ? 0 : cs_snip_add(cs, hash, line, nr_lines);
```

Comment:
Once you use `expect()` you can avoid this and just pass through again.

Reply:
`cs_snip_add` resizes the store and increments `nr_snips` by +1. We don't really want that in the dedup case.

Unless by "pass through" you meant passing through the return value `ret` as is.